packages/core/src/lib/code-snippet.svelte

```diff
@@ -0,0 +1,227 @@
+import cx from 'classnames';
+
+import type { IconName } from '$lib';
+import IconButton from './button/icon-button.svelte';
+
+/**
+ * The language to use for syntax highlighting. Must be a language supported by
+ * prism: https://prismjs.com/#supported-languages
+ */
+export let language: string;
+
+/**
+ * The code to render in the code-snippet.
+ */
+export let code: string;
+
+/**
+ * The theme to use for syntax highlighting. Must be a theme provided by
+ * prism-themes: https://github.com/PrismJS/prism-themes#available-themes
+ *
+ * Currently only the `vs` and `vsc-dark-plus` themes are supported.
+ */
+export let theme: CodeSnippetTheme = 'vs';
+
+/**
+ * Whether or not to show a copy-to-clipboard button.
+ */
+export let showCopyButton = true;
+
+/**
+ * Some prism languages have dependencies. For example, C++ requires C. If the
+ * passed `language` has dependencies, they must be included here to be loaded.
+ *
+ * See: https://prismjs.com/plugins/autoloader/
+ */
+export let dependencies: string[] = [];
+
+/**
+ * We use the prism autoloader to handle loading in language grammar files. The
+ * default path for the grammar files is a CDN link so we don't have to include
+ * grammar files in our bundle. If you prefer to point to your own grammars and
+ * bypass the CDN, define the path to those files.
+ *
+ * See: https://prismjs.com/plugins/autoloader/
+ */
+export let grammarsPath = `https://cdnjs.cloudflare.com/ajax/libs/prism/${PRISM_VERSION}/components/`;
+
+/** Additional CSS classes to pass to the aside. */
+let extraClasses: cx.Argument = '';
+export { extraClasses as cx };
+
+let element: HTMLElement | undefined;
+
+$: copyButtonLabel = 'Copy';
+$: copyButtonIcon = 'content-copy' as IconName;
+
+const dispatch = createEventDispatcher<{
+  copy: { succeeded: boolean; message: string };
+}>();
+
+const copyToClipboard = async () => {
+  try {
+    await navigator.clipboard.writeText(code);
+
+    copyButtonLabel = 'Copied';
+    copyButtonIcon = 'check';
+
+    dispatch('copy', {
+      succeeded: true,
+      message: 'Successfully copied snippet to the clipboard',
+    });
+  } catch {
+    copyButtonLabel = 'Failed';
+    copyButtonIcon = 'alert';
+
+    dispatch('copy', {
+      succeeded: false,
+      message: 'Failed to copy snippet to the clipboard',
+    });
+  }
+
+  window.setTimeout(() => {
+    copyButtonLabel = 'Copy';
+    copyButtonIcon = 'content-copy';
+  }, 2000);
+};
+
+const highlight = () => {
+  if (element) {
+    /**
+     * We need to reset the inner HTML of the element to the raw value of
+     * `code` so it can rescanned for highlighting from the raw.
+     */
+    element.innerHTML = code;
+    highlightElement(element);
+  }
+};
+
+$: {
+  if (code) {
+    highlight();
+  }
+}
+
+onMount(async () => {
+  try {
+    /**
+     * Load the themes, for vite to properly discover them we cannot use string templates,
+     * so we use a switch statement for our supported themes.
+     */
```

Comment:
We may be able to use [`import.meta.glob`](https://vitejs.dev/guide/features.html#glob-import) to help

Reply:
I believe I tried that originally which is where I discovered the issue with vite but I can try again.

Reply:
Yeah in that doc it says:

> You should also be aware that all the arguments in the import.meta.glob must be passed as literals. You can NOT use variables or expressions in them.

If we tried to use the theme as a variable I am pretty sure it will still yell at us.

Reply:
import.meta.glob returns an object, keyed by module name. So you'd use the variable when accessing the module via the object, not when constructing the glob. That being said, I don't remember if it plays nicely with node_modules, any probably isn't worth the effort

Reply:
It actually might, I was trying to query it by the file name but I think it is the full module path that is the key. I am testing it out now to see.